server.js

```diff
@@ -0,0 +1,204 @@
+var config = require('./config')
+var log = require('./log')
+var subscriptions = require('./subscriptions')
+var notifier = require('./notifier')
+var hapi = require('hapi')
+var joi = require('joi')
+
+var db = require('./db/' + config.db.driver)
+
+var server = new hapi.Server()
+server.connection({
+  host: config.server.host,
+  port: config.server.port
+})
+
+server.route([
+  {
+    method: 'POST',
+    path: '/v1/publish',
+    config: {
+      validate: {
+        payload: {
+          events: joi.array().max(1000).includes(joi.string()).required()
+        }
+      }
+    },
+    handler: function (req, reply) {
+      var events = req.payload.events
+      var notifyUrls = {}
+      for (var i = 0; i < events.length; i++) {
+        try {
+          var event = JSON.parse(events[i])
+          db.append(event)
+          subscriptions.whoToNotify(event).forEach(
```

Comment:
And I'm happy to keep it since you've already got it here, but we could consider pulling this push/webhook stuff from v0 if we need to, since I don't think we have any Q1 reliers who want it.